addons/account/models/account_move.py

```diff
@@ -1153,7 +1153,7 @@ def _get_last_sequence_domain(self, relaxed=False):
         param = {'journal_id': self.journal_id.id}
 
         if not relaxed:
-            domain = [('journal_id', '=', self.journal_id.id), ('id', '!=', self.id or self._origin.id), ('name', 'not in', ('/', False))]
+            domain = [('journal_id', '=', self.journal_id.id), ('move_type', '=', self.move_type), ('id', '!=', self.id or self._origin.id), ('name', 'not in', ('/', False))]
```

Comment:
You need to add that new clause iff `self.journal_id.refund_sequence` (like done on line 1168)
This is probably why runbot is red too

Reply:
I agree with you, I have adapted but the test still fails.

After investigation, I think the test is not correct: 

https://github.com/odoo/odoo/blob/7a7dc212af7279544cffba362bce7daa4fc00865/addons/account/tests/test_reconciliation_matching_rules.py#L744

--> we are posting a move_type 'entry' in a sales journal (NB: `self.journal_id.refund_sequence` will still be `True` ), which does not seem allowed based on some tests made on Runbot.

Should I change the default journal to 'default_journal_misc' instead?

Reply:
They change made should be something like
```python
if self.journal_id.refund_sequence:
    refund_types = ('out_refund', 'in_refund')
    domain+= [('move_type', 'in' if self.move_type in refund_types else 'not in', refund_types)]
```
Do you have something similar?

Reply:
But the test is indeed weird. At first glance what you do here should not break it; but the test is indeed wrong and needs to be changed.